### PR TITLE
Resolves #880: add information on property updates to the change recorder

### LINKED ISSUE #880: Enhancement: add information on property updates to the change recorder
Change recorder only records dependency updates. In this enhancement I'll add property updates. This implies a new version of the XSD and a new namespace:

```xml
<schema xmlns="http://www.w3.org/2001/XMLSchema"
        elementFormDefault="qualified"
        xmlns:u="http://www.mojohaus.org/versions-maven-plugin/schema/updates/2.0"
        targetNamespace="http://www.mojohaus.org/versions-maven-plugin/schema/updates/2.0">

    <complexType name="VersionChangeType"/>

    <complexType name="DependencyVersionChangeType">
        <complexContent>
            <extension base="u:VersionChangeType">
                <attribute name="kind" type="string" use="required"/>
                <attribute name="groupId" type="string" use="required"/>
                <attribute name="artifactId" type="string" use="required"/>
                <attribute name="oldVersion" type="string" use="required"/>
                <attribute name="newVersion" type="string" use="required"/>
            </extension>
        </complexContent>
    </complexType>

    <complexType name="PropertyVersionChangeType">
        <complexContent>
            <extension base="u:VersionChangeType">
                <attribute name="property" type="string" use="required"/>
                <attribute name="oldValue" type="string" use="required"/>
                <attribute name="newValue" type="string" use="required"/>
            </extension>
        </complexContent>
    </complexType>

    <element name="updates">
        <complexType>
            <sequence maxOccurs="unbounded">
                <choice>
                    <element name="dependencyUpdate" type="u:DependencyVersionChangeType"/>
                    <element name="propertyUpdate" type="u:PropertyVersionChangeType"/>
                </choice>
            </sequence>
        </complexType>
    </element>

</schema>
```

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Release notes:
- new version and new namespace for the change recorder
- change recorder also records property changes

### CODE REVIEW

**Comment by ajarmoniuk** [comment]
@slawekjaranowski please review

**Comment by ajarmoniuk** [comment]
target is 2.15

File: versions-common/src/main/java/org/codehaus/mojo/versions/change/DefaultPropertyVersionChange.java
Comment (slawekjaranowski) [inline versions-common/src/main/java/org/codehaus/mojo/versions/change/DefaultPropertyVersionChange.java:4]: Not ASF project - please use standard header for new files

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: versions-api/src/main/java/org/codehaus/mojo/versions/api/change/VersionChange.java
Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/change/VersionChange.java:24]: Hmm, that's true. Ok, I'll change DepenencyVersionChange back to VersionChange and this one to Change

File: versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/DependencyChangeRecord.java
Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/DependencyChangeRecord.java:26]: Right, if we treat this as API then it's pretty hard to do any modifications here. We can extend, but it's not going to make much sense for properties.

File: versions-api/src/main/java/org/codehaus/mojo/versions/api/change/VersionChange.java
Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/change/VersionChange.java:24]: As above. So, I'll think about it.

Comment (slawekjaranowski) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/change/VersionChange.java:24]: I think that it is still a new API, I hope that a not many developer use it. 
So it can be time to change.

File: versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java
Comment (slawekjaranowski) [inline versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: What for is a return value?

Comment (slawekjaranowski) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: there was a no question ... :-)

Comment (slawekjaranowski) [inline versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: What do you think add two methods

```
void recordChange(DependencyChangeRecord changeRecord);

void recordChange(PropertyChangeRecord changeRecord);
```

Implementation will be easier - we will not need to use `instance of`

Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: Well, this has its downsizes too -- this means a change in the interface.

Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: I'll remove the generic void recordChange(ChangeRecord changeRecord);

Comment (ajarmoniuk) [reply versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java]: Applied; not squashing yet, waiting for final review.

File: versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java
Comment (slawekjaranowski) [inline versions-api/src/main/java/org/codehaus/mojo/versions/api/recording/ChangeRecorder.java:39]: dependency -> property

File: versions-common/src/main/java/org/codehaus/mojo/versions/change/DefaultPropertyVersionChange.java
Comment (slawekjaranowski) [reply versions-common/src/main/java/org/codehaus/mojo/versions/change/DefaultPropertyVersionChange.java:4]: Why resolved?

File: versions-common/src/main/java/org/codehaus/mojo/versions/recording/DefaultPropertyChangeRecord.java
Comment (slawekjaranowski) [inline versions-common/src/main/java/org/codehaus/mojo/versions/recording/DefaultPropertyChangeRecord.java:10]: Not ASF.
Please put a license header after package ...

**Review by slawekjaranowski — COMMENTED** [review]
We need update example in https://www.mojohaus.org/versions/versions-api/change-recorder.html

**Comment by ajarmoniuk** [comment]
Applied the review comments.

File: versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy
Comment (slawekjaranowski) [inline versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy:2]: eh... once again it is not AFS project - new files should have not such headers

Please check other new files also.

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy:2]: Oops. Will do.

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy:2]: By the way, what do you think of introducing the rat plugin?

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy:2]: Edit: I've just checked, there's really **plenty** of files with the old license. Will create a separate PR to update those.

Comment (ajarmoniuk) [reply versions-maven-plugin/src/it/it-changerecord-update-parent-001/verify.groovy]: @slawekjaranowski phew, updated the changed files only. For the rest a separate PR should be created.

**Pull request merged**
